server/utils/files/journal-io.ts

```
@@ -0,0 +1,249 @@
+  }
+}
+
+// ── Daily file listing ──────────────────────────────────────────
+
+export interface DailyFileEntry {
+  year: string;
+  month: string;
+  day: string;
+}
+
+export async function listDailyFiles(r?: string): Promise<DailyFileEntry[]> {
```

ESLint

[error] Refactor this function to reduce its Cognitive Complexity from 21 to the 15 allowed.

(sonarjs/cognitive-complexity)